tests/conftest.py

```diff
@@ -70,36 +27,36 @@ def ca_ssl_context(cert_authority: trustme.CA) -> ssl.SSLContext:
     return ctx
 
 
-@pytest.fixture(scope="session")
-def example_org_cert(cert_authority: trustme.CA) -> trustme.LeafCert:
-    return cert_authority.issue_cert("example.org")
+def wait_until_pproxy_serve_on_port(host: str, port: int):
+    while True:
+        try:
+            sock = socket.create_connection((host, port))
+        except ConnectionRefusedError:
+            time.sleep(0.25)
+        else:
+            sock.close()
+            break
 
 
 @pytest.fixture(scope="session")
-def example_org_cert_path(example_org_cert: trustme.LeafCert) -> typing.Iterator[str]:
-    with example_org_cert.private_key_and_cert_chain_pem.tempfile() as tmp:
-        yield tmp
+def proxy_server() -> typing.Iterator[URL]:
```

Comment:
Instead of overriding the existing code, which makes things _a bit_ hard to review (nothing blocking though), should we take the opportunity here and define any code related to the test proxy in `conftest/utils.py`? I think it's a good pattern to isolate anything that doesn't _need_ to be defined in `conftest.py` to somewhere else, for readability. Bonus point here is the diff should be easier to review. :-)

Reply:
I hope I understood you correctly. I've moved all the code apart from fixture initialization into the `tests/utils.py`